pysimplesoap/transport.py

```diff
@@ -139,6 +144,128 @@ def request(self, url, method="GET", body=None, headers={}):
+#
+try:
+    from urllib3 import PoolManager, __version__
+    from urllib3.exceptions import HTTPError
+    from urllib3.util import make_headers
+except ImportError:
+    pass
+else:
+
+    class urllib3Transport(TransportBase):
+
+        """ Provides methods for sending http requests (specifically SOAP requests)
+        with connection pooling
+
+        Attributes:
+            num_pools (int): The maximum number of connections in the pool (default: 10)
+            timeout (float): Set the http request timeout in seconds
+            headers (dict): (optional) Headers to include with all requests
+            maxsize (int): The maximum number of connection in the pool
+            retries (int): The number of retries if the http request fails
+        """
+
+        _wrapper_version = "urllib3 %s" % __version__
+
+        _wrapper_name = 'urllib3'
+
+        def __init__(self, timeout=None, proxy=None, cacert=None, 
+                     sessions=None, num_pools=None, maxsize=None, retries=None,
+                     cert_file=None, ca_certs=None, key_file=None, 
+                     cert_reqs=None, headers=None):
+            if num_pools:
+                try:
+                    num_pools = int(num_pools)
+                except ValueError:
+                    raise ValueError("num_pools parameter must be an integer")
+
+            if timeout:
+                try:
+                    timeout = float(timeout)
+                except ValueError:
+                    raise ValueError("timeout parameter must be a float")
+
+            if maxsize:
+                try:
+                    maxsize = int(maxsize)
+                except ValueError:
+                    raise ValueError("maxsize parameter must be an integer")
+
+            if retries:
+                try:
+                    retries = int(retries)
+                except ValueError:
+                    raise ValueError("retries parameter must be an integer")
+
+            self.http_headers = {}
+
+            self.manager = PoolManager(
+                timeout=timeout,
+                num_pools=num_pools,
+                maxsize=maxsize,
+                retries=retries,
+                headers=headers)
+
+        def request(self, url, method="GET", body=None, headers={}):
+            """ Makes an HTTP request from the connection pool
+
+            Params:
+                url (str): The url requested
+                method (str): The method for request (e.g. GET, POST, etc)
+                body (dict): Data being sent
+                headers (dict): Headers
+
+            Returns:
+                tuple: A tuple containing the headers (dict) and the return data (str)
+
+            Raises:
+                urllib3.exceptions.HTTPError
+            """
+            headers.update(self.http_headers)
+            r = self.manager.urlopen(method, url, headers=headers, body=body)
+
+            if r.status in (200, 201, 202, 203, 204, 205, 206):
```

Comment:
Soapfaults may come back with status code 500. The payload should be parsed. Can you use a similar logic as urllib2 here?